src/app/pages/request-query/request-query.component.ts

```diff
@@ -0,0 +1,148 @@
+})
+export class RequestQueryComponent implements OnInit, IModelPage {
+  dataRequest?: DataRequest;
+  queryType: DataRequestQueryType = 'none';
+  dataElements?: DataElementSearchResult[];
+  query?: DataRequestQuery;
+  condition: QueryCondition = defaultQueryCondition;
+  status: 'init' | 'loading' | 'ready' | 'error' = 'init';
+  dirty = false;
+  original = '';
+
+  constructor(
+    private route: ActivatedRoute,
+    private dataRequests: DataRequestsService,
+    private toastr: ToastrService,
+    private broadcast: BroadcastService
+  ) {}
+
+  isDirty(): boolean {
+    return this.dirty;
+  }
+
+  ngOnInit(): void {
+    this.route.params
+      .pipe(
+        switchMap((params) => {
+          this.queryType = params.queryType;
+
+          const requestId: Uuid = params.requestId;
+          this.status = 'loading';
+          return this.dataRequests.get(requestId);
+        }),
+        catchError(() => {
+          this.toastr.error('There was a problem finding your request.');
+          this.status = 'error';
+          return EMPTY;
+        }),
+        switchMap((dataRequest) => {
+          if (!dataRequest || !dataRequest.id) {
+            return EMPTY;
+          }
+          this.dataRequest = dataRequest;
+
+          return forkJoin([
+            this.dataRequests.listDataElements(dataRequest),
+            this.dataRequests.getQuery(dataRequest.id, this.queryType),
+          ]);
+        }),
+        catchError(() => {
+          this.toastr.error('There was a problem finding your request queries.');
+          this.status = 'error';
+          return EMPTY;
+        })
+      )
+      .subscribe(([dataElements, query]) => {
+        this.dataElements = this.mapDataElements(dataElements);
+        this.query = query;
+        this.condition = this.query ? this.query.condition : defaultQueryCondition;
+        this.original = JSON.stringify(this.condition);
+        this.status = 'ready';
+      });
+  }
+
+  onQueryChange(value: QueryCondition) {
+    const change = JSON.stringify(value);
+    this.dirty = this.original !== change;
+  }
+
+  save() {
+    if (!this.dataRequest || !this.dataRequest.id) {
+      return;
+    }
+
+    const payload: DataRequestQueryPayload = {
+      ruleId: this.query?.ruleId,
+      representationId: this.query?.representationId,
+      type: this.queryType,
+      condition: this.condition,
+    };
+
+    this.broadcast.loading({ isLoading: true, caption: 'Saving your query...' });
+    this.dataRequests
+      .createOrUpdateQuery(this.dataRequest.id, payload)
+      .pipe(
+        catchError(() => {
+          this.toastr.error('There was a problem saving your query.');
+          return EMPTY;
+        }),
+        finalize(() => this.broadcast.loading({ isLoading: false }))
+      )
+      .subscribe((submitted) => {
+        this.toastr.success(`Your ${this.queryType} query was saved successfully.`);
+        // TODO: change below to a redirect
```

Comment:
If the query is saved successfully, I don't think a redirect should then take place. A user may wish to save as they go, not be instantly moved elsewhere as soon as they click the save button.

Reply:
Changed this and the tests to reflect what you've asked for.

Reply:
This all looks good now. Ready for merge.